lib/ansible/cli/__init__.py

```diff
@@ -7,17 +7,36 @@
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
+import sys
+
+# Used for determining if the system is running a new enough python version
+# and should only restrict on our documented minimum versions
+if sys.version_info < (3, 8):
+    raise SystemExit(
+        'ERROR: Ansible requires Python 3.8 or newer on the controller. '
+        'Current version: %s' % ''.join(sys.version.splitlines())
+    )
+
+import errno
 import getpass
 import os
 import subprocess
-import sys
-
+import traceback
 from abc import ABCMeta, abstractmethod
+from pathlib import Path
+
+try:
+    from ansible import constants as C
+    from ansible.utils.display import Display, initialize_locale
+    initialize_locale()
+    display = Display()
+except Exception as e:
+    print('ERROR: %s' % e, file=sys.stderr)
+    sys.exit(5)
```

Comment:
Mind putting this magic value into a constant with a descriptive name?
I'd read much better if it was
```suggestion
    sys.exit(RC_CLI_INIT_FAILURE)
```
or something like that.

Reply:
I think I might defer this to another PR later.  I don't necessarily exactly know what rc 5 is supposed to indicate.  I just copied it from the previous file.  I know there is another PR attempting to better define the exit codes.